hash-parser/src/parse.rs

```diff
@@ -0,0 +1,22 @@
+//! Hash compiler module for converting from tokens to an AST tree
+//
+// All rights reserved 2021 (c) The Hash Language authors
+use crate::ast::*;
+use crate::error::ParseError;
+use crate::grammar::{HashParser, Rule};
+
+// use pest::iterators::{Pair, Pairs};
+use pest::Parser;
+
+pub fn module(source: &str) -> Result<Module, ParseError> {
+    let result = HashParser::parse(Rule::module, source);
+
+    match result {
+        Ok(_pairs) => {
+            // here is where we do all the ast-ification
+            let vec = Vec::new();
+            Ok(Module { contents: vec })
+        }
+        Err(err) => Err(ParseError::from(err)),
+    }
```

Comment:
This can be shortened to
```rust
let _pairs = HashParser::parse(Rule::module, source)?;
Ok(Module { contents: vec![] })
```

Reply:
well, since we will be doing stuff with the rules when we write code to convert into AST, I think it is sensible to leave it like this.